cmd/todd-agent/main.go

```diff
@@ -51,6 +51,11 @@ func main() {
 
 	cfg := config.GetConfig(arg_config)
 
+	// Create optDir
+	if err := os.MkdirAll(cfg.LocalResources.OptDir, 0755); err != nil {
```

Comment:
Honestly I haven't gotten to the distribution question. My thought is that once we reach a more stable "beta" status (working on publishing those milestones now actually) we can address it. For now I'd like to focus on developers, and if there are some savvy operators that need some help with make, that's fine with me.

I created https://github.com/Mierdin/todd/issues/26 to track the packaging idea, because I agree this is one of the hotpoints for getting ToDD usable, especially for non-developers. In the meantime, however, I'd like to see all of the setup tasks (i.e. setting up /etc/todd and /opt/todd) remain in the makefile, and keep that as the primarily install method. For the time being, to use ToDD, you need to compile from source. The makefile should make this fairly easy for now (basically ```make``` then ```sudo make install``` is the idea).

I obviously love all of the changes that make ToDD more Go-idiomatic from a structure perspective (i.e. creating the "cmd" directory, etc) so those should stay.